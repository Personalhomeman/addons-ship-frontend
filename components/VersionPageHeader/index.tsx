import { ReactNode } from 'react';

import css from './style.scss';
import SVG from 'react-svg';

interface VersionPageHeaderProps {
  appName: string;
  versionIconUrl: string;
  children: ReactNode;
}

export default ({ appName, versionIconUrl, children }: VersionPageHeaderProps) => (
  <div className={css.versionPageHeader}>
    <a href="/" className={css.appName}>
      {appName}
    </a>
    <div className={css.versionNameWrapper}>
      <SVG src={versionIconUrl} className={css.versionIcon} />
      <h2 className={css.versionName}>{children}</h2>
    </div>
  </div>
);
