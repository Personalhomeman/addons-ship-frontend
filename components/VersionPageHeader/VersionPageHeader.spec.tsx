import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import VersionPageHeader from '.';

describe('VersionPageHeader', () => {
  it('renders correctly', () => {
    const tree = toJSON(shallow(<VersionPageHeader appName='My app' versionIconUrl='https://www.bitrise.io/assets/svg/logo-bitrise.svg'>My app version 1.0</VersionPageHeader>));
    expect(tree).toMatchSnapshot();
  });
});
