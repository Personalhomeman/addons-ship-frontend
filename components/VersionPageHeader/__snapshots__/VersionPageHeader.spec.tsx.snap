// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionPageHeader renders correctly 1`] = `
<div
  className="versionPageHeader"
>
  <a
    className="appName"
    href="/"
  >
    My app
  </a>
  <div
    className="versionNameWrapper"
  >
    <InlineSVG
      baseURL=""
      cacheGetRequests={false}
      className="versionIcon"
      onLoad={[Function]}
      src="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
      supportTest={[Function]}
      uniquifyIDs={true}
      wrapper={[Function]}
    />
    <h2
      className="versionName"
    >
      My app version 1.0
    </h2>
  </div>
</div>
`;
